coverage/jsonreport.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+# Licensed under the Apache License: http://www.apache.org/licenses/LICENSE-2.0
+# For details: https://github.com/nedbat/coveragepy/blob/master/NOTICE.txt
+
+"""Json reporting for coverage.py"""
+import json
+import sys
+import time
+
+from coverage import __version__
+from coverage.report import get_analysis_to_report
+from coverage.results import Numbers
+
+
+class JsonReporter(object):
+    """A reporter for writing JSON coverage results."""
+
+    def __init__(self, coverage):
+        self.coverage = coverage
+        self.config = self.coverage.config
+        self.total = Numbers()
+        self.report_data = {}
+        self.data = self.coverage.get_data()
+        self.has_arcs = self.data.has_arcs()
+
+    def report(self, morfs, outfile=None):
+        """Generate a json report for `morfs`.
+
+        `morfs` is a list of modules or file names.
+
+        `outfile` is a file object to write the json to
+
+        """
+        outfile = outfile or sys.stdout
+        self.data.set_query_contexts(self.config.report_contexts)
+        meta = {
+            "version": __version__,
+            "timestamp": str(int(time.time())),
```

Comment:
Would ISO8601 be more friendly?

Reply:
I think in my head I was thinking `time.time()` would be easier for machine parsing. However its not like ISO8601 would be hard... and it has the benefit of being much more human readable. So sure thing

Reply:
Actually looking at this it may have been me doing what xml was doing. Changed to `isoformat`